ci/scripts/python_build.bat

```diff
@@ -0,0 +1,139 @@
+@rem Licensed to the Apache Software Foundation (ASF) under one
+@rem or more contributor license agreements.  See the NOTICE file
+@rem distributed with this work for additional information
+@rem regarding copyright ownership.  The ASF licenses this file
+@rem to you under the Apache License, Version 2.0 (the
+@rem "License"); you may not use this file except in compliance
+@rem with the License.  You may obtain a copy of the License at
+@rem
+@rem   http://www.apache.org/licenses/LICENSE-2.0
+@rem
+@rem Unless required by applicable law or agreed to in writing,
+@rem software distributed under the License is distributed on an
+@rem "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+@rem KIND, either express or implied.  See the License for the
+@rem specific language governing permissions and limitations
+@rem under the License.
+
+@echo on
+
+set SOURCE_DIR=%1
+set CMAKE_INSTALL_PREFIX=%2
+set CPP_SOURCE_DIR=%SOURCE_DIR%\cpp
+set CPP_BUILD_DIR=%SOURCE_DIR%\build
+echo C++ source dir is %CPP_SOURCE_DIR%
+
+echo Building for Windows ...
+
+@REM List installed Pythons
+py -0p
+
+%PYTHON_CMD% -m sysconfig || exit /B 1
+
+@REM Setup MSVC environment
+
+call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+@echo on
+
+echo "=== CCACHE Stats before build ==="
+ccache -sv
+
+echo "=== Building Arrow C++ libraries ==="
+set ARROW_ACERO=ON
+set ARROW_DATASET=ON
+set ARROW_FLIGHT=OFF
+set ARROW_GANDIVA=OFF
+set ARROW_GCS=OFF
+set ARROW_HDFS=ON
+set ARROW_MIMALLOC=ON
+set ARROW_ORC=OFF
+set ARROW_PARQUET=ON
+set PARQUET_REQUIRE_ENCRYPTION=ON
+set ARROW_SUBSTRAIT=ON
+set ARROW_S3=ON
+set ARROW_TENSORFLOW=ON
+set ARROW_WITH_BROTLI=ON
+set ARROW_WITH_BZ2=OFF
+set ARROW_WITH_LZ4=ON
+set ARROW_WITH_SNAPPY=ON
+set ARROW_WITH_ZLIB=ON
+set ARROW_WITH_ZSTD=ON
+set CMAKE_BUILD_TYPE=Release
```

Comment:
Oh, do we want to make that Debug perhaps? That will enable some assertions on the C++ side.

Reply:
Perhaps as a separate issue/PR.

Reply:
Ah, I see. Perhaps that should raise an error instead of silencing changing the build type. Anyway, it's for another issue :)

Reply:
Oh, by the way, that check might be obsolete because:
> Changed in version 3.8: Release builds and debug builds are now ABI compatible

https://docs.python.org/3/using/configure.html#python-debug-build